packages/server-core/src/projects/project/project.class.ts

```diff
@@ -382,160 +556,87 @@ export class Project extends Service {
-          sourceBranch: data.sourceBranch,
-          updateType: data.updateType,
-          updateSchedule: data.updateSchedule,
-          updateUserId: userId
-        })
-
-    returned.needsRebuild = typeof data.needsRebuild === 'boolean' ? data.needsRebuild : true
-
-    if (!existingProjectResult) {
-      await this.app.service('project-permission').create({
-        projectId: returned.id,
-        userId
-      })
-    }
-
-    if (returned.name !== projectName)
-      await super.patch(existingProjectResult.id, {
-        name: projectName
-      })
-
-    if (data.reset) {
-      let repoPath = await getAuthenticatedRepo(githubIdentityProvider.oauthToken, data.destinationURL)
-      if (!repoPath) repoPath = data.destinationURL //public repo
-      await git.addRemote('destination', repoPath)
-      await git.raw(['lfs', 'fetch', '--all'])
-      await git.push('destination', branchName, ['-f', '--tags'])
-      const { commitSHA, commitDate } = await this._getCommitSHADate(projectName)
-      await super.patch(returned.id, {
-        commitSHA,
-        commitDate
-      })
-    }
-    // run project install script
-    if (projectConfig.onEvent) {
-      await onProjectEvent(
-        this.app,
-        projectName,
-        projectConfig.onEvent,
-        existingProjectResult ? 'onUpdate' : 'onInstall'
+      } catch (err) {
+        console.log('Old update job did not exist, continuing...')
+      }
+      const job = await k8BatchClient.createNamespacedJob(
+        'default',
+        getProjectUpdateJobBody(data, image, params!.user!.id)
       )
+      let counter = 0
+      const jobFinishedPromise = new Promise((resolve, reject) => {
+        const interval = setInterval(async () => {
+          counter++
+          const jobLabelSelector = `etherealengine/projectField=${data.name},etherealengine/release=${process.env.RELEASE_NAME}`
+
+          const updateJob = await k8BatchClient.listNamespacedJob(
+            'default',
+            undefined,
+            false,
+            undefined,
+            undefined,
+            jobLabelSelector
+          )
+          if (updateJob && updateJob.body.items.length > 0) {
+            const succeeded = updateJob.body.items.filter((item) => item.status && item.status.succeeded === 1)
+            const failed = updateJob.body.items.filter((item) => item.status && item.status.failed === 1)
+            if (succeeded.length > 0 || failed.length > 0) clearInterval(interval)
+            if (succeeded.length > 0) resolve(null)
+            if (failed.length > 0) reject()
+          }
+          if (counter >= UPDATE_JOB_TIMEOUT) {
+            clearInterval(interval)
+            reject('Project update timed out; try again later or check error logs of update job')
+          }
+        }, 1000)
+      })
+      try {
+        await jobFinishedPromise
+        const returned = await this.app.service('project').Model.findOne({
+          where: {
+            [Op.or]: [
```

Comment:
Do we need this `or` here? As there is just 1 condition.

Reply:
Simplified this to generic feathers syntax.